core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -4169,6 +4169,43 @@ public void testSeqNoGenerator() throws IOException {
         }
     }
 
+    public void testWriteIndexingBuffer() throws IOException {
```

Comment:
can we add a version of this test that also looks at the live version size and falls into a full refresh?  I also don't see anywhere a test that shows that the live version map memory usage is reflected in the InternalEngine#getIndexBufferRAMBytesUsed method. Maybe I miss it though (as it is an existing problem).